source/opt/common_uniform_elim_pass.cpp

```diff
@@ -130,6 +130,51 @@ ir::Instruction* CommonUniformElimPass::GetPtr(
   return ptrInst;
 }
 
+bool CommonUniformElimPass::IsVolatileStructType(const ir::Instruction& AccessChainInst) {
+  assert(AccessChainInst.opcode() == SpvOpAccessChain);
+  uint32_t memberTypeId = AccessChainInst.GetSingleWordOperand(3);
+  // OpConstant
```

Comment:
You can only know if this is a constant if you've proven that the base pointer has a struct type.  You should check that first;  return false before this point if it's not a struct type.

Now, it's true that the code for Vulkan will always have a struct as the type of a UBO or SSBO.    But I think OpenGL's use allows "loose" uniforms, so you should guard against that here.